internal/stores/hostdb.go

```diff
@@ -741,3 +771,29 @@ func insertAnnouncements(tx *gorm.DB, as []announcement) error {
 	}
 	return tx.Create(&hosts).Error
 }
+
+func updateRevisionNumberAndHeight(db *gorm.DB, fcid types.FileContractID, revisionHeight, revisionNumber uint64) error {
+	return updateActiveAndArchivedContract(db, fcid, map[string]interface{}{
+		"revision_height": revisionHeight,
+		"revision_number": fmt.Sprint(revisionNumber),
+	})
+}
+
+func updateProofHeight(db *gorm.DB, fcid types.FileContractID, blockHeight uint64) error {
+	return updateActiveAndArchivedContract(db, fcid, map[string]interface{}{
+		"proof_height": blockHeight,
+	})
+}
+
+func updateActiveAndArchivedContract(db *gorm.DB, fcid types.FileContractID, updates map[string]interface{}) error {
```

Comment:
`nit` should we rename `db` to `tx` to signal that the intention is to do this inside of a transaction?